env/lib/python3.12/site-packages/pip/_internal/models/selection_prefs.py

```
@@ -0,0 +1,51 @@
+    """
+    Encapsulates the candidate selection preferences for downloading
+    and installing files.
+    """
+
+    __slots__ = [
+        "allow_yanked",
+        "allow_all_prereleases",
+        "format_control",
+        "prefer_binary",
+        "ignore_requires_python",
+    ]
```

Ruff (0.13.3)

`SelectionPreferences.__slots__` is not sorted

Apply a natural sort to `SelectionPreferences.__slots__`

(RUF023)